app/src/lib/dispatcher/app-store.ts

```diff
@@ -1875,13 +1891,40 @@ export class AppStore {
     return this.refreshGitHubRepositoryInfo(repository)
   }
 
+  private async getAccountForRemoteURL(
+    remote: string
+  ): Promise<IGitAccount | null> {
+    const gitHubRepository = matchGitHubRepository(this.accounts, remote)
+    if (gitHubRepository) {
+      const account = getAccountForEndpoint(
+        this.accounts,
+        gitHubRepository.endpoint
+      )
+      if (account) {
+        return account
+      }
+    }
+
+    const hostname = getGenericHostname(remote)
+    const username = await getGenericUsername(hostname)
```

Comment:
A thought: could we do something in `localStorage` rather than falling back to `keytar` for tracking the host-username lookup work? We're still using `process.env.DESKTOP_USERNAME` to pass this through ASKPASS, which means maybe we can rely on it like we do for the rest of the GitHub account info...